google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerTableExporter.java

```diff
@@ -55,47 +60,90 @@ public SpannerTableExporter(SpannerDialect spannerDialect) {
+          initCommands.stream()
+              .filter(cmd -> cmd instanceof ReplaceInitCommand)
+              .collect(Collectors.toList());
+    } else if (initCommands.stream().anyMatch(cmd -> cmd instanceof ReplaceInitCommand)) {
+      // Only ReplaceInitCommands, but there is nothing to replace, so we return early.
+      return;
+    }
+    for (InitCommand initCommand : initCommands) {
+      addStatementAfterDdlBatch(metadata, initCommand.getInitCommands());
+    }
   }
 
   @Override
   public String[] getSqlDropStrings(Table currentTable, Metadata metadata) {
     initializeUniqueConstraints(currentTable);
-    return buildSqlStrings(currentTable, metadata, Action.DROP);
+    return buildSqlStrings(currentTable, metadata, Action.DROP).toArray(new String[0]);
   }
 
   /**
    * Initializes the table exporter for if a new create-table or drop-table sequence is starting.
    */
   public void init(
-      Metadata metadata,
-      SpannerDatabaseInfo spannerDatabaseInfo,
-      Action schemaAction) {
+      Metadata metadata, SpannerDatabaseInfo spannerDatabaseInfo, Action schemaAction) {
     tableDependencyTracker.initializeDependencies(metadata, schemaAction);
     spannerTableStatements.initializeSpannerDatabaseInfo(spannerDatabaseInfo);
   }
 
-  private String[] buildSqlStrings(Table currentTable, Metadata metadata, Action schemaAction) {
+  private List<String> buildSqlStrings(Table currentTable, Metadata metadata, Action schemaAction) {
     Collection<Table> tablesToProcess = tableDependencyTracker.getDependentTables(currentTable);
 
-    List<String> ddlStatements = tablesToProcess.stream()
-        .flatMap(table -> {
-          if (schemaAction == Action.CREATE) {
-            return spannerTableStatements.createTable(table, metadata).stream();
-          } else {
-            return spannerTableStatements.dropTable(table).stream();
-          }
-        })
+    return tablesToProcess.stream()
+        .flatMap(
+            table -> {
+              if (schemaAction == Action.CREATE) {
+                return spannerTableStatements.createTable(table, metadata).stream();
+              } else {
+                return spannerTableStatements.dropTable(table).stream();
+              }
+            })
         .collect(Collectors.toList());
+  }
+
+  static void addStatementAfterDdlBatch(Metadata metadata, String[] statements) {
+    // Find the RunBatchDdl auxiliary object which can run statements after the DDL batch.
+    Optional<RunBatchDdl> runBatchDdl =
+        metadata.getDatabase().getAuxiliaryDatabaseObjects().stream()
+            .filter(obj -> obj instanceof RunBatchDdl)
+            .map(obj -> (RunBatchDdl) obj)
+            .findFirst();
 
-    return ddlStatements.toArray(new String[ddlStatements.size()]);
+    if (runBatchDdl.isPresent()) {
+      for (String statement : statements) {
+        runBatchDdl.get().addAfterDdlStatement(statement);
+      }
+    } else {
+      throw new IllegalStateException(
```

Comment:
Can you add a test to throw an exception here?

Reply:
Done (although it is a situation that should never happen, so the test is slightly simple/silly)